src/controllers/validator/service-create-request.ts

```diff
@@ -0,0 +1,70 @@
+import type { IValidationResult, IValidator, Validatable } from './validator.js';
+import { Helpers, type IHelpers } from './helpers.js';
+import type { CreateDIDService } from '../../types/shared.js';
+
+// It's a special validator for the service create request. It's not a part of the DID Document
+export class CreateDIDDocumentServiceValidator implements IValidator {
+	helpers: IHelpers;
+
+	constructor(helpers?: IHelpers) {
+		if (!helpers) {
+			helpers = new Helpers();
+		}
+		this.helpers = helpers;
+	}
+
+	validateEach(service: CreateDIDService): IValidationResult {
+		if (!service.idFragment) {
+			return {
+				valid: false,
+				error: `service.id is required in object ${service}`,
+			};
+		}
+		if (!service.type) {
+			return {
+				valid: false,
+				error: `service.type is required in object ${service}`,
+			};
+		}
+		if (!service.serviceEndpoint) {
+			return {
+				valid: false,
+				error: `service.serviceEndpoint is required in object ${service}`,
+			};
+		}
+		if (!Array.isArray(service.serviceEndpoint)) {
+			return {
+				valid: false,
+				error: `service.serviceEndpoint should be an array in object ${service}`,
+			};
+		}
+		return { valid: true };
+	}
+
+	validate(services: Validatable): IValidationResult {
+		services = services as CreateDIDService[];
+		if (!Array.isArray(services)) {
+			return {
+				valid: false,
+				error: 'Service value for Create DID Request should be an array',
+			};
+		}
+		const ids = services.map((s) => s.idFragment);
+		if (!this.helpers.isUnique(ids)) {
+			return {
+				valid: false,
+				error: 'Service for Create DID Request does not have unique service.',
+			};
+		}
+
+		const results = services.map((s) => this.validateEach(s));
+		if (results.some((r) => !r.valid)) {
+			return {
+				valid: false,
+				error:
+					'Service for Create DID Request has validation errors: ' + results.map((r) => r.error).join(', '),
+			};
+		}
+		return { valid: true };
+	}
```

Comment:
We have 2 kind of Service structure to be honest on CredentialService side.
1. Came from `did-resolver` and it's used as a part of did document.
2. Another one is only our own and instead of `id` field it uses `idFragment` one and the main purpose of it - just provide the template for compiling the service section when the DID will be generated. For building DID we firstly make DID body generation and when the DID will be known we create service section